kernel/core/utils/stringtools.cpp

```diff
@@ -989,6 +990,40 @@ static bool isDateTimePattern(const std::string& sStr, size_t pos)
 }
 
 
+/////////////////////////////////////////////////
+/// \brief This function is used during the detectCommonType Function to
+///        count detected Number Formats troughout one Column
+///
+/// \param numType int
+/// \return void
+///
+/////////////////////////////////////////////////
+static void voteNumType(int numType)  {
+    if((numType & NUM_DECIMAL_EU && numType & NUM_K_US) ||
+       (numType & NUM_DECIMAL_US && numType & NUM_K_EU)) {
+        num_format_votes[5]++;  // INVALID Combination

```

Comment:
You do not want to use number literals here. There might be furthe formats in the future. Consider a solution like so:

```C++
enum
{
    NUM_DECIMAL_EU,
    // all other format types
    NUM_FORMAT_COUNT // This will be the number of formats available (and therefore scale with further formats)
}

// the declaration of the necessary array like so (will scale automatically)
int num_format_votes[NUM_FORMAT_COUNT];